sycl/include/sycl/detail/cg_types.hpp

```diff
@@ -128,13 +128,29 @@ class NDRDescT {
     Dims = Dims_;
   }
 
+  template <int Dims_> void setClusterDimensions(sycl::range<Dims_> N) {
+    if (Dims_ != Dims) {
+      throw std::runtime_error(
+          "Dimensionality of cluster, global and local ranges must be same");
+    }
+
+    for (int I = 0; I < Dims_; ++I) {
+      ClusterDimensions[I] = N[I];
+    }
+
+    for (int I = Dims; I < 3; ++I) {
+      ClusterDimensions[I] = 1;
+    }
+  }
+
   sycl::range<3> GlobalSize;
   sycl::range<3> LocalSize;
   sycl::id<3> GlobalOffset;
   /// Number of workgroups, used to record the number of workgroups from the
   /// simplest form of parallel_for_work_group. If set, all other fields must be
   /// zero
   sycl::range<3> NumWorkGroups;
+  sycl::range<3> ClusterDimensions;
```

Comment:
Why not just default initialize this
```suggestion
  sycl::range<3> ClusterDimensions{1, 1, 1};
```
And then remove

Reply:
I see none of the other ranges are initialized in that fashion, hence I followed suite. Was it intentional or just a happenstance ?

Reply:
I don't know why the others are initialized in `setNDRangeLeftover`. I would prefer this method that I have mentioned